### PR TITLE
Compatibility with local assets and better handling of deposits XCM fee

### PR DESCRIPTION
**Pull request by fxgamundi**
### Checklist

- [x] If this requires a documentation change, I have created a PR in [moonbeam-docs](https://github.com/PureStake/moonbeam-docs) repository.
- [x] If this requires it, I have updated the Readme
- [x] If necessary, I have updated the examples
- [x] I have verified if I need to create/update unit tests
- [x] I have verified if I need to create/update acceptance tests

### CODE REVIEW

**Review by ekenigs — CHANGES_REQUESTED** [review]
I also tried to deposit xusdc with our dapp and it didn't work. Not sure what could be wrong.

**Comment by fxgamundi** [comment]
> I also tried to deposit xusdc with our dapp and it didn't work. Not sure what could be wrong.

Centrifuge still haven't enabled the TransferMultiCurrencies extrinsic in their chain, so there's no way to actually test USDC right now

File: packages/sdk/src/sdk/sdk.deposit.ts [hunk 1]
```diff
@@ -148,54 +140,83 @@ export async function getDepositData<
...
     existentialDeposit,
     min,
-    moonChainFee,
+    moonChainFee: {
+      balance: xcmFeeBalance,
+      decimals: xcmFeeAssetConfig.asset.isNative
+        ? meta.decimals
+        : xcmFeeDecimals,
+      fee: xcmFee,
+      isDifferentAsset: xcmFeeAssetConfig.asset.id !== asset.id,
```
Comment (ekenigs) [inline packages/sdk/src/sdk/sdk.deposit.ts:214]: I'm not a big fan of this flag. I would not add it.

**Review by ekenigs — CHANGES_REQUESTED** [review]
Looks good besides the flag. Please check create a PR to docs because interface change.

**Comment by fxgamundi** [comment]
[Moonbeam docs PR](https://github.com/PureStake/moonbeam-docs/pull/441)

**Review by ekenigs — APPROVED** [review]

**Comment by fxgamundi** [comment]
Updated to latest PolkadotJS deps and tested every deposit against live version. All good

**Review by ekenigs — APPROVED** [review]

**Pull request merged**